Add reads2graph

Jappy0 opened the pull request:
reads2graph is a practical method for constructing an edit-distance-based read graph for a short-read sequencing dataset via minimizer and order min hash bucketing and graph traversal.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

Jappy0 commented:
@BiocondaBot please add label

Hi there,

I was trying to create a new pull request for my new recipe reads2graph, but my previous commit of the recipe deepdirect mixed up with this new recipe. Could you please help me divide them? And I don't want to update deepdirect with that commit (update deepdirect to 0.2.5). 

Thanks so much for your help.

Best,

Jappy

martin-g commented on recipes/deepdirect/meta.yaml:
```diff
@@ -1,39 +1,35 @@
-    - keras >=2.4.0
-    - colorlog ==6.7.0
-    - tqdm >=4.64.1
+    - python >=3.8
```
ditto

Jappy0 commented on recipes/deepdirect/meta.yaml:
```diff
@@ -1,39 +1,35 @@
 {% set name = "deepdirect" %}
-{% set version = "0.2.4" %}
+{% set version = "0.2.5" %}
```
Hi, 
Yes. You are right. deepdirect is not related to this PR. reads2graph does not depend on deepdirect.

If I remember correctly, deepdirect was committed last year. And the deepdirect 0.2.5 had been released to bioconda last year. I have no idea why it still exists.

martin-g replied:
https://anaconda.org/bioconda/deepdirect indeed says that the latest version is 0.2.5
Please undo the changes in this file for this PR.
I something needs to be updated in deepdirect then please do it in a separate PR

Jappy0 commented on recipes/deepdirect/meta.yaml:
```diff
@@ -1,39 +1,35 @@
 requirements:
   host:
-    - python
+    - python >=3.8
```
Ok, got it. Thanks so much.

martin-g commented on recipes/reads2graph/meta.yaml:
```diff
@@ -0,0 +1,51 @@
+build:
+  number: 0
+  script: build.sh
+  noarch: generic  # Noarch because it's a C++ software built from source
```
You resolved the conversation but actually this line must be removed.
A compiled software is not a `noarch`.

Jappy0 replied:
Yes, you are right; I apologise for any inconvenience. It has been a long time since I have used this, and I may forget many things.

Jappy0 commented:
Hi @martin-g,

I think I made some mistakes when I added the reads2graph. Now I have removed the irrelevant commits to this PR and should have resolved the errors you pointed out. Thanks so much for your help. Please review it again.

Best regards,

Jappy

Jappy0 commented:
@BiocondaBot rebuild

martin-g commented on recipes/reads2graph/meta.yaml:
```diff
@@ -0,0 +1,49 @@
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage("reads2graph", max_pin="x") }}
```
```suggestion
    - {{ pin_subpackage(name, max_pin="x") }}
```

martin-g commented on recipes/reads2graph/meta.yaml:
```diff
@@ -0,0 +1,49 @@
+    - cmake >=3.21
+    - gxx_linux-64 ==13.2.0
+  host:
+    - seqan3 ==3.3.0
```
I am not a big fan of exact version pins but I guess you have your reasons.

martin-g commented:
There are Git conflicts that should be resolved.

Jappy0 commented:
> There are Git conflicts that should be resolved.

Hi Martin, 

Yes, the Git conflicts should be resolved. Thanks so much for the help. But I may have to trouble you again as the bioconda-recipes test process has been stuck with "Expected — Waiting for status to be reported" for almost 24 hours. There may be some errors in my recipes. I have no idea how to fix this issue.

Best regards,

Jappy

martin-g commented:
No idea what is going on with Azure CI...
I guess the easiest for you is to close this PR and create a fresh new one!

Jappy0 commented:
> No idea what is going on with Azure CI... I guess the easiest for you is to close this PR and create a fresh new one!

Got it. Let me have a try.
